Objects/object.c

```diff
@@ -1699,12 +1694,13 @@ _PyObject_GenericSetAttrWithDict(PyObject *obj, PyObject *name,
 
     if (dict == NULL) {
         PyObject **dictptr;
-        if ((tp->tp_flags & Py_TPFLAGS_INLINE_VALUES) && _PyObject_InlineValues(obj)->valid) {
-            res = _PyObject_StoreInstanceAttribute(
-                    obj, _PyObject_InlineValues(obj), name, value);
+
+        if ((tp->tp_flags & Py_TPFLAGS_INLINE_VALUES)) {
+            res = _PyObject_TryStoreInstanceAttribute(obj, name, value);
```

Comment:
Why does this have the `Try` prefix? Can it fail in ways that a "normal" function like `PyDict_SetItem` would not fail?

Reply:
Nope, this is a hold over from the previous version where it returned a value indicating the caller needed to do the store to the dict directly.